improver_tests/developer_tools/test_MOMetadataInterpreter.py

```diff
@@ -0,0 +1,505 @@
+    """Test the interpreter can distinguish between the Global Model and the
+    Global Grid in cube titles"""
+    probability_above_cube.attributes[
+        "title"
+    ] = "UKV Model Forecast on Global 10 km Standard Grid"
+    interpreter.run(probability_above_cube)
+    assert interpreter.model == "UKV"
+
+
+def test_probabilities_below(blended_probability_below_cube, interpreter):
+    """Test interpretation of blended probability of max temperature in hour
+    below threshold"""
+    interpreter.run(blended_probability_below_cube)
+    assert interpreter.prod_type == "gridded"
+    assert interpreter.field_type == "probabilities"
+    assert interpreter.diagnostic == "air_temperature"
+    assert interpreter.relative_to_threshold == "less_than"
+    assert interpreter.methods == " maximum over time"
+    assert interpreter.post_processed
+    assert interpreter.model == "UKV, MOGREPS-UK"
+    assert interpreter.blended
+    assert not interpreter.warnings
+
+
+def test_snow_level(snow_level_cube, interpreter):
+    """Test interpretation of a diagnostic cube with "probability" in the name,
+    which is not designed for blending with other models"""
+    interpreter.run(snow_level_cube)
+    assert interpreter.prod_type == "gridded"
+    assert interpreter.field_type == "realizations"
+    assert (
+        interpreter.diagnostic == "probability_of_snow_falling_level_below_ground_level"
+    )
+    assert interpreter.relative_to_threshold is None
+    assert not interpreter.methods
+    assert interpreter.post_processed
+    assert interpreter.model is None
+    assert not interpreter.blended
+    assert not interpreter.warnings
+
+
+def test_spot_median(blended_spot_median_cube, interpreter):
+    """Test interpretation of spot median"""
+    interpreter.run(blended_spot_median_cube)
+    assert interpreter.prod_type == "spot"
+    assert interpreter.field_type == "percentiles"
+    assert interpreter.diagnostic == "air_temperature"
+    assert interpreter.relative_to_threshold is None
+    assert not interpreter.methods
+    assert interpreter.post_processed
+    assert interpreter.model == "UKV, MOGREPS-UK"
+    assert interpreter.blended
+    assert not interpreter.warnings
+
+
+def test_wind_direction(wind_direction_cube, interpreter):
+    """Test interpretation of wind direction field with mean over realizations
+    cell method"""
+    interpreter.run(wind_direction_cube)
+    assert interpreter.diagnostic == "wind_from_direction"
+    assert interpreter.model == "MOGREPS-UK"
+    assert not interpreter.blended
+
+
+def test_weather_code(wxcode_cube, interpreter):
+    """Test interpretation of weather code field"""
+    interpreter.run(wxcode_cube)
+    assert interpreter.diagnostic == "weather_code"
+    assert interpreter.model == "UKV, MOGREPS-UK"
+    assert interpreter.blended
+
+
+## Test errors and warnings (input cubes in alphabetical order by fixture)
```

Comment:
I wonder whether these tests could be ordered or grouped a bit better, so they more clearly line up with the private functions they are testing. Maybe we just need a few more comments in saying things like 
```
# testing attribute errors
# testing cell method errors
# testing probabilistic metadata errors
```

Reply:
I did discuss the order with Caroline and she rearranged them into what is currently present. I'd be happy for the extra comments to be included though.

Reply:
I'm happy with whatever reordering and comments make it clearest for the user.  It's a pretty complicated set of cases, and worth noting that some cases are diagnostic-specific, so you will find this tricky and moderately messy regardless.  The only thing I would like specifically to keep is having all the "success" cases together at the start of the file.